Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -1874,6 +1878,93 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     }
 }
 
+// MARK: webRequest
+
+bool WebExtensionContext::hasPermissionToSendWebRequestEvent(WebExtensionTab *tab, const URL& resourceURL, const ResourceLoadInfo& loadInfo)
+{
+    if (!tab)
+        return false;
+
+    if (!hasPermission(_WKWebExtensionPermissionWebRequest, tab))
+        return false;
+
+    auto tabURL = tab->url();
+    if (tabURL.isValid() && !hasPermission(tabURL, tab))
+        return false;
+
+    if (resourceURL.isValid() && !hasPermission(resourceURL, tab))
```

Comment:
Is `isValid()` needed? `hasPermission()` already does an `isEmpty()` check and returns `false`.

Reply:
I missed that it did the isEmpty check. I think we can get rid of the isValid check.

Reply:
Actually - is the URL is empty, `hasPermission` will return false. We want to return true in that case, so I want to keep the check.